lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1412,6 +1412,40 @@ public function get_attribute( $name ) {
 		return html_entity_decode( $raw_value );
 	}
 
+	/**
+	 * Returns the names of all attributes matching a given prefix in the currently-opened tag.
+	 *
+	 * Example:
+	 * <code>
+	 *     $p = new WP_HTML_Tag_Processor( '<div data-enabled class="test" data-test-id="14">Test</div>' );
+	 *     $p->next_tag( [ 'class_name' => 'test' ] ) === true;
+	 *     $p->get_attribute_names_with_prefix() === array( 'data-enabled', 'data-test-id' );
+	 *
+	 *     $p->next_tag( [] ) === false;
+	 *     $p->get_attribute_names_with_prefix() === null;
+	 * </code>
+	 *
+	 * @since 6.2.0
+	 *
+	 * @param string $prefix Prefix of requested attribute names.
+	 * @return array|null List of attribute names, or `null` if not at a tag.
+	 */
+	function get_attribute_names_with_prefix( $prefix ) {
+		if ( $this->is_closing_tag || null === $this->tag_name_starts_at ) {
+			return null;
```

Comment:
I think I like this, but I want to make sure we aren't overlooking the possibility of wanting to return `array()` when not on a tag. It feels like the kind of thing where I might have suggested the opposite: return `null` if we're not on a tag 🙃 but oh well - is there a reason to differentiate between "there are no attributes matching this prefix" and "you shouldn't be calling this here"?

Reply:
Yeah, good point. I don't feel strongly about this -- it used to return `array()` prior to this commit.

I think my chief reason to change it to `null` was consistency with both the "not-on-a-tag" case, and `get_attribute()`'s behavior when on a closing tag (for which I even added a test case, 8c51859ddc4eadb1ae483cd692f0b41f966247c6) -- thinking that `null` made sense as the general indicator for "no attributes here".

But an empty array might make just as much sense.

> is there a reason to differentiate between "there are no attributes matching this prefix" and "you shouldn't be calling this here"

To push this further, would it then make sense to return `array()` also if `null === $this->tag_name_starts_at`?

Reply:
it doesn't seem like it'd matter to me to differentiate why you didn't call it in the right place, only that you _did_ call it from the wrong place.

if we return `null` we know people will end up writing code that crashes because they won't check the results. if we return an empty array we know that people will introduce data corruption by assuming it worked.

while it may fluctuate on a weekly basis, I think given this thought I lean towards `null` which at least gives a way to know more clearly if something is missing attributes or that we're looking at the wrong spot in the document.

Reply:
Happy to stick with `null` 😊 👍